zerver/lib/actions.py

```diff
@@ -4518,6 +4533,24 @@ def do_update_message(user_profile: UserProfile, message: Message,
         )
         changed_messages += messages_list
 
+        if new_stream is not None:
+            assert stream_being_edited is not None
+            UserMessage.objects.filter(
+                user_profile__in=[sub.user_profile for sub in guest_subs],
+                message__in=changed_messages,
+            ).delete()
+
+            delete_event: Dict[str, Any] = {
+                'type': 'delete_message',
+                'message_ids': [msg.id for msg in changed_messages],
+                'message_type': 'stream',
+                'stream_id': stream_being_edited.id,
+                'topic': orig_topic_name,
+            }
```

Comment:
Can you add a prep commit defining a TypedDict for the `delete_event` format?  We generally try hard to avoid two code paths creating the same event type without strong assurances that they'll be the same format; we've had some bad experiences with accidentally partial migrations in the past.